app/routes.py

```diff
@@ -1,50 +1,37 @@
 from app import app, BUILD_DIR
-from app.controllers import (HomeController, LoginController,
-                             LoginPageController, OMDBSearchController)
+from app.controllers import (
+    LoginController,
+    OMDBSearchController,
+)
 from app.models import User
-from app.session_util import login_required
+from app.session_util import is_user_logged_in, login_required
 
-from flask import jsonify, send_from_directory
+from flask import jsonify, send_from_directory, session
 
 
 # For all routes, return index.html (so that React Router can handle routing).
 # Flask matches the most specific routes first, so this will only be called
 # if nothing else matches.
-# TODO: Add all login_required logic back
 @app.route('/', defaults={'path': ''})
 @app.route('/<path:path>')
 def serve(path):
     return send_from_directory(BUILD_DIR, 'index.html')
 
 
-@app.route('/current-user')
-def get_current_user():
-    user = User.query.first()
-    user_dict = {'email': user.email} if user else None
-    return jsonify(user=user_dict)
+@app.route('/session', methods=['GET'])
+def is_session_valid():
+    if is_user_logged_in():
+        return jsonify(is_session_active=True, email=session['email'])
+    else:
+        return jsonify(is_session_active=False, email='')
```

Comment:
I wonder if `is_session_active` and `email` should be the same thing (e.g. the presence of a `session_email` implies that there is an active session, otherwise it's not).

On the one hand, it kind of conflates two concerns, but on the other hand, it removes a class of bug where e.g. `is_active_session` is set to `True` but there's no email

Reply:
(And it means that you don't always have to check both properties, etc)